core/src/main/scala/cats/Selective.scala

```diff
@@ -0,0 +1,78 @@
+package cats
+
+import simulacrum.{noop, typeclass}
+import scala.annotation.implicitNotFound
+
+@implicitNotFound("Could not find an instance of Selective for ${F}")
+@typeclass trait Selective[F[_]] extends Applicative[F] {
```

Comment:
Just to reiterate, I'm strongly -1 on adding a typeclass whose superclass is a valid implementation.

I believe that is what we have here unless I'm mistaken.

I think we should remove this and instead just move all these methods up to Applicative if you are really intent on keeping select weak.

Reply:
I'm relying heavily Andrey's paper for inspiration, but my hesitation is not Because Haskell™.  The problem here is two principles we both agree on are in tension:

1. No typeclasses that can be implemented by a superclass.
2. Laws that overrides equate with their base implementations.  

* The status quo violates principle #1.
* We can default `select == selectA`, but that's incompatible with rigidity, violating principle #2.
* Moving an abstract `select` down to `Apply` breaks compatibility.
* Not introducing `select` before rigidity eschews important functionality.
* `Selective` could just not participate in this hierarchy, but then we can't enforce rigidity of monads by law.

I'm leaning toward `Apply.select == selectA`, but no law.  Principle #2 above is violated, but something has to yield here.

Reply:
Okay, thank you for clarifying this. I think we have never violated 1, but I think we may have violated 2 without knowing since we didn't write the laws previously. For instance ap2 uses a strange order in the default implementation. I think many people won't notice if they don't use it, or use it with a commutative monad.

I tend to agree with your final result: provide a default implementation for select == selectA, but no law about it. The law I would like is that Monad.select == selectM

Reply:
`select == selectM` is already added to `MonadLaws`, so everyone agrees here.  And `ap == apS` looks like it could apply to all `FlatMap`.  I can't think of any `FlatMap`s for which that's untrue.

So then what is `RigidSelective` and not `FlatMap`?  Well, `Validated` can avoid evaluating the `ff` effect, but isn't associative when `ap == apS`.  The only two unflatmappable rigids I can think of are `Over` and a free selective, both of which are interesting, neither of which I'm convinced belong in cats-core.  And `Over` often runs with `Under`: it's the canonical example of why `select == selectA` is a fine default but a bad law.

I propose that the rigidity laws move to `FlatMap`, and no new typeclass is introduced until a need for a non-flatmapping rigidity constraint is demonstrated.